src/lib.rs

```diff
@@ -624,4 +814,89 @@ mod tests {
             Err(_) => (),
         }
     }
+
+    #[test]
+    fn cascade_builder_test_collision() {
+        let mut builder =
+            CascadeBuilder::new(HashAlgorithm::Sha256, b"naclnaclnacl".to_vec(), 1, 1);
+        builder.include(b"collision!".to_vec());
+        builder.exclude(b"collision!".to_vec());
+        assert!(builder.finalize().is_err());
+    }
+
+    #[test]
+    #[should_panic(expected="capacity violation")]
+    fn cascade_builder_test_exclude_too_few() {
+        let mut builder =
+            CascadeBuilder::new(HashAlgorithm::Sha256, b"naclnaclnacl".to_vec(), 1, 1);
+        builder.include(b"1".to_vec());
+        assert!(builder.finalize().is_err());
```

Comment:
Having both the `assert!` and `should_panic` seems conflicting, or at least redundant.